pkg/datapath/linux/node.go

```diff
@@ -630,6 +645,75 @@ func (n *linuxNodeHandler) insertNeighbor(newNode *nodeTypes.Node, ifaceName str
 	}
 }
 
+// Must be called with linuxNodeHandler.mutex held.
+func (n *linuxNodeHandler) refreshNeighbor(nodeToRefresh *nodeTypes.Node, ifaceName string) {
+	if nodeToRefresh.IsLocal() {
+		return
+	}
+
+	nodeIP := nodeToRefresh.GetNodeIP(false).To4()
+	nextHopIPv4 := make(net.IP, len(nodeIP))
+	copy(nextHopIPv4, nodeIP)
+
+	scopedLog := log.WithFields(logrus.Fields{
+		logfields.Interface: ifaceName,
+		logfields.IPAddr:    nextHopIPv4,
+	})
+
+	srcIPv4, nextHopIPv4 := n.getSrcAndNextHopIPv4(nextHopIPv4, ifaceName)
+	if srcIPv4 == nil {
+		return
+	}
+
+	nextHopStr := nextHopIPv4.String()
+	n.neighNextHopByNode[nodeToRefresh.Identity()] = nextHopStr
+	oldNeigh, oldNeighFound := n.neighByNextHop[nextHopStr]
+	_, refCountExists := n.neighNextHopRefCount[nextHopStr]
+
+	// nextHop hasn't been arpinged before OR the arping failed
+	//if n.neighNextHopRefCount.Add(nextHopStr) || !found {
+	linkAttr, err := netlink.LinkByName(ifaceName)
+	if err != nil {
+		scopedLog.WithError(err).Error("Failed to retrieve iface by name (netlink)")
+		return
+	}
+	link := linkAttr.Attrs().Index
+
+	hwAddr, err := arp.PingOverLink(linkAttr, srcIPv4, nextHopIPv4)
+	if err != nil {
+		scopedLog.WithError(err).Error("arping failed")
+		return
+	}
+
+	// MAC address not changed.
+	if oldNeighFound && hwAddr.String() == oldNeigh.HardwareAddr.String() {
+		return
+	}
+
+	scopedLog = scopedLog.WithField(logfields.HardwareAddr, hwAddr)
+
+	neigh := netlink.Neigh{
+		LinkIndex:    link,
+		IP:           nextHopIPv4,
+		HardwareAddr: hwAddr,
+		State:        netlink.NUD_PERMANENT,
+	}
+	if err := netlink.NeighSet(&neigh); err != nil {
+		scopedLog.WithError(err).Error("Failed to replace neighbor entry")
+		return
+	}
+	scopedLog.Info("Neighbor MAC address has changed, replaced neighbor entry")
```

Comment:
Worst case, this line will log every 30s in each agent for each node in the cluster. If I got that right, then this would pollute the logs with unhelpful information at a very frequent rate. I'd suggest we downgrade this to debug level. 

Plus, in the case that we can't reach a node due to a stale MAC address, we can check the controller status via `cilium status --verbose`. If the refresh controller failed at some point, that would be an indication to check the logs for an error, which the code logs anyway. That should be enough really. We want to know if something has gone wrong, rather than a status update. 

I'm sure this was helpful for debugging though :)